### PR TITLE
Upgrade all libraries and required nodejs version  

### PR DESCRIPTION
**Pull request by robsonpeixoto**
This is a big PR, my bad 😞 , that:
- remove the `node_modules` directory
- upgrade all dependencies (`16 vulnerabilities (6 moderate, 4 high, 6 critical)`)
- upgrade to node 16
- use vercel/ncc to package the `.js` file
- remove not used configs from `tsconfig.json`
- remove not used dependency

This github-action still have some problems:
- using deprecated lib: https://www.npmjs.com/package/request
- code is not formatted
- coveralls dep has a lot of critical security problems: `29 vulnerabilities (1 low, 8 moderate, 14 high, 6 critical)`

Everything was copied from https://github.com/actions/typescript-action

### CODE REVIEW

**Comment by jrfnl** [comment]
Duplicate of #137 ?

**Comment by franky47** [comment]
Might as well use Node 18 for actions, as it's now the current LTS.

**Comment by robsonpeixoto** [comment]
Only node12 and node16 are supported. Reference: https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runsusing

**Comment by robsonpeixoto** [comment]
@jrfnl is not duplicate of https://github.com/coverallsapp/github-action/pull/137, because this PR is a big refactor on the build process.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,8 @@
 .DS_Store
 .idea
 coverage
+# Dependency directory
+node_modules
+node_modules/
```
Comment (paescuj) [inline .gitignore:6]: No need to add that twice
```suggestion
```

**Comment by paescuj** [comment]
> upgrade all dependencies

Did you forget to commit the `package.json` & `package-lock.json` files?

File: .gitignore (hunk 1 above)
Comment (robsonpeixoto) [reply .gitignore:5]: Thanks, done in 5ab0f8013fedd6c7fb2b078ce8353867ceb04eab

Comment (paescuj) [reply .gitignore:5]: Thanks!

**Comment by robsonpeixoto** [comment]
> Did you forget to commit the package.json & package-lock.json files?
@paescuj, read the commits below:

- https://github.com/coverallsapp/github-action/pull/139/commits/ce512ee8bd2786f3433ad2d11f1770752dabdf02
- https://github.com/coverallsapp/github-action/pull/139/commits/3e3d30c9e85967a48e2223de89063465059a97b2

File: src/run.ts [hunk 2]
```diff
@@ -96,14 +96,7 @@ export async function run() {
...
 
-    let file;
-
-    try {
-      file = fs.readFileSync(pathToLcov, 'utf8');
-    } catch (err) {
-      throw new Error("Lcov file not found.");
-    }
-
+    let file = await fs.readFile(pathToLcov, 'utf8');
```
Comment (paescuj) [inline src/run.ts:99]: Wouldn't it be nicer to keep the `try...catch` statement here?

**Comment by paescuj** [comment]
> > Did you forget to commit the package.json & package-lock.json files?
> > @paescuj, read the commits below:
> 
>     * [ce512ee](https://github.com/coverallsapp/github-action/commit/ce512ee8bd2786f3433ad2d11f1770752dabdf02)
> 
>     * [3e3d30c](https://github.com/coverallsapp/github-action/commit/3e3d30c9e85967a48e2223de89063465059a97b2)

Ah, great! Sorry, it didn't show up in the diff because it's limited to 3.000 files :-)

**Comment by robsonpeixoto** [comment]
Sorry! Delete the `node_modules` created a huge diff.

**Comment by paescuj** [comment]
> Sorry! Delete the `node_modules` created a huge diff.

Definitely not your fault 😃

**Comment by fluxsauce** [comment]
This is amazing, thank you for working on this @robsonpeixoto !

**Comment by paulrberg** [comment]
Any ETA on when this PR will be reviewed and merged by the @coveralls team?

**Comment by robsonpeixoto** [comment]
ping @nickmerwin @anjin

File: .gitignore (hunk 1 above)
Comment (robsonpeixoto) [reply .gitignore:5]: Just updating the commit: https://github.com/coverallsapp/github-action/pull/139/commits/418ea71351dbca27a4fc48c1e1d2ad8a2d3ff32a

File: src/run.ts (hunk 2 above)
Comment (robsonpeixoto) [reply src/run.ts]: Done: 17ed4c213e51a88ba30b8eacf81ff3cdd1da036c

Comment (paescuj) [reply src/run.ts]: Nice, thanks! 🥇

**Comment by CoreyGahafer** [comment]
@nickmerwin or @anjin can this be merged in?

**Comment by nopeless** [comment]
Hoping this will be merged soon so I don't have to see warnings :)

**Comment by szepeviktor** [comment]
@afinetooth Maybe we need someone to maintain this repo once a month. What do you think?

**Pull request closed without merging**